src/app/script/app_object.cpp

```diff
@@ -498,6 +498,44 @@ int App_useTool(lua_State* L)
   return 0;
 }
 
+int App_tip(lua_State* L)
+{
+  const auto* ctx = App::instance()->context();
+  if (!ctx || !ctx->isUIAvailable() || !StatusBar::instance())
+    return 0; // No UI to show the tooltip
+
+  std::string text;
+  double duration = 2.0;
+
+  if (lua_istable(L, 1)) {
+    int type = lua_getfield(L, 1, "text");
+    if (type == LUA_TSTRING)
+      text = lua_tostring(L, -1);
+    lua_pop(L, 1);
+
+    type = lua_getfield(L, 1, "duration");
+    if (type == LUA_TNUMBER)
+      duration = lua_tonumber(L, -1);
+    lua_pop(L, 1);
+  }
+  else {
+    if (!lua_isstring(L, 1))
+      return luaL_error(L, "app.tip text parameter must be a string");
+
+    text = lua_tostring(L, 1);
+
+    if (lua_isnumber(L, 2))
+      duration = lua_tonumber(L, 1);
```

Comment:
It looks like a typo
```suggestion
    if (lua_isnumber(L, 2))
      duration = lua_tonumber(L, 2);
```